Calibration/IsolatedParticles/plugins/IsolatedGenParticles.cc

```diff
@@ -16,55 +16,332 @@
+  std::vector<double> *t_photonEneR;
+  std::vector<double> *t_eleEneR;
+  std::vector<double> *t_muEneR;
+
+  std::vector<double> *t_maxNearPIsoR;
+  std::vector<double> *t_cHadronEneIsoR, *t_cHadronEneIsoR_1, *t_cHadronEneIsoR_2, *t_cHadronEneIsoR_3;
+  std::vector<double> *t_nHadronEneIsoR;
+  std::vector<double> *t_photonEneIsoR;
+  std::vector<double> *t_eleEneIsoR;
+  std::vector<double> *t_muEneIsoR;
+
+  std::vector<double> *t_maxNearPHCR;
+  std::vector<double> *t_cHadronEneHCR, *t_cHadronEneHCR_1, *t_cHadronEneHCR_2, *t_cHadronEneHCR_3;
+  std::vector<double> *t_nHadronEneHCR;
+  std::vector<double> *t_photonEneHCR;
+  std::vector<double> *t_eleEneHCR;
+  std::vector<double> *t_muEneHCR;
+
+  std::vector<double> *t_maxNearPIsoHCR;
+  std::vector<double> *t_cHadronEneIsoHCR, *t_cHadronEneIsoHCR_1, *t_cHadronEneIsoHCR_2, *t_cHadronEneIsoHCR_3;
+  std::vector<double> *t_nHadronEneIsoHCR;
+  std::vector<double> *t_photonEneIsoHCR;
+  std::vector<double> *t_eleEneIsoHCR;
+  std::vector<double> *t_muEneIsoHCR;
+
+  std::vector<int>    *t_L1Decision;
+  std::vector<double> *t_L1CenJetPt,    *t_L1CenJetEta,    *t_L1CenJetPhi;
+  std::vector<double> *t_L1FwdJetPt,    *t_L1FwdJetEta,    *t_L1FwdJetPhi;
+  std::vector<double> *t_L1TauJetPt,    *t_L1TauJetEta,    *t_L1TauJetPhi;
+  std::vector<double> *t_L1MuonPt,      *t_L1MuonEta,      *t_L1MuonPhi;
+  std::vector<double> *t_L1IsoEMPt,     *t_L1IsoEMEta,     *t_L1IsoEMPhi;
+  std::vector<double> *t_L1NonIsoEMPt,  *t_L1NonIsoEMEta,  *t_L1NonIsoEMPhi;
+  std::vector<double> *t_L1METPt,       *t_L1METEta,       *t_L1METPhi;
+
+  spr::genSimInfo isoinfo1x1,   isoinfo3x3,   isoinfo7x7,   isoinfo9x9,   isoinfo11x11;
+  spr::genSimInfo isoinfo15x15, isoinfo21x21, isoinfo25x25, isoinfo31x31;
+  spr::genSimInfo isoinfoHC1x1, isoinfoHC3x3, isoinfoHC5x5, isoinfoHC7x7;
+  spr::genSimInfo isoinfoR,     isoinfoIsoR,  isoinfoHCR,   isoinfoIsoHCR;
+
+};
 
 IsolatedGenParticles::IsolatedGenParticles(const edm::ParameterSet& iConfig) {
 
-  genSrc_    = iConfig.getUntrackedParameter("GenSrc",std::string("generatorSmeared"));
+  usesResource("TFileService");
+
+  genSrc_     = iConfig.getUntrackedParameter("GenSrc",std::string("generatorSmeared"));
+  useHepMC_   = iConfig.getUntrackedParameter<bool>("UseHepMC", false );
+  ptMin_      = iConfig.getUntrackedParameter<double>("PTMin", 1.0);
+  etaMax_     = iConfig.getUntrackedParameter<double>("MaxChargedHadronEta", 2.5);
+  a_coneR_    = iConfig.getUntrackedParameter<double>("ConeRadius",34.98);
+  a_charIsoR_ = a_coneR_ + 28.9;
+  a_neutIsoR_ = a_charIsoR_*0.726;
+  a_mipR_     = iConfig.getUntrackedParameter<double>("ConeRadiusMIP",14.0);
+  a_Isolation_= iConfig.getUntrackedParameter<bool>("UseConeIsolation",false);
+  pCutIsolate_= iConfig.getUntrackedParameter<double>("PMaxIsolation",20.0);
+  verbosity_  = iConfig.getUntrackedParameter<int>("Verbosity", 0);
+  debugL1Info_= iConfig.getUntrackedParameter<bool>( "DebugL1Info", false );
```

Comment:
this could go into the initializer list of the constructor

Reply:
Done